scene/3d/cpu_particles_3d.cpp

```diff
@@ -880,7 +880,9 @@ void CPUParticles3D::_particles_process(double p_delta) {
 				} break;
 				case EMISSION_SHAPE_RING: {
 					real_t ring_random_angle = Math::randf() * Math_TAU;
-					real_t ring_random_radius = Math::sqrt(Math::randf() * (emission_ring_radius - emission_ring_inner_radius * emission_ring_inner_radius) + emission_ring_inner_radius * emission_ring_inner_radius);
```

Comment:
Wasn't it just missing a single `* emission_ring_radius` (besides the added inner radius clamping)? :thinking:

Like:
```cpp
real_t ring_random_radius = Math::sqrt(Math::randf() * (emission_ring_radius * emission_ring_radius - emission_ring_inner_radius * emission_ring_inner_radius) + emission_ring_inner_radius * emission_ring_inner_radius);
```
or:
```cpp
real_t w = Math::randf();
real_t ring_random_radius = Math::sqrt(w * emission_ring_radius * emission_ring_radius + (1.0f - w) * emission_ring_inner_radius * emission_ring_inner_radius);
```

Reply:
Argh, yes you are right. it just needed to be squared as well. I'll do a new commit with this correct. Thanks Kleonc. and sorry for the delay on this.